### PR TITLE
Elaborate on VM resource in architecture document

### PR DESCRIPTION
**Pull request by ravidbro**
Signed-off-by: Ravid Brown <ravid@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dhiller** [comment]
I overrode due to ppc lane instability.

File: docs/architecture.md [hunk 1]
```diff
@@ -85,6 +85,17 @@ And a simplified version:
...
     * …
+  * KubeVirt Custom Resources
+    * VirtualMachine (VM) Foo
+        -> VirtualMachineInstance (VMI) Foo
+    * VirtualMachineInstanceReplicaSet (VMIRS) Bar
+        -> VirtualMachineInstance (VMI) Bar
+
+VirtualMachineInstance (VMI) are the custom resource that represents the basic ephemeral building block of an instance.
+In a lot of cases this object won't be created directly by the user but by a high level resource.
+High level resources fot VMI can be:
```
Comment (dhiller) [inline docs/architecture.md]:
```suggestion
High level resources for VMI can be:
```

File: docs/architecture.md [hunk 2]
```diff
@@ -85,6 +85,17 @@ And a simplified version:
     * VMI Foo
     * VMI Bar
     * …
+  * KubeVirt Custom Resources
+    * VirtualMachine (VM) Foo
+        -> VirtualMachineInstance (VMI) Foo
+    * VirtualMachineInstanceReplicaSet (VMIRS) Bar
+        -> VirtualMachineInstance (VMI) Bar
+
+VirtualMachineInstance (VMI) are the custom resource that represents the basic ephemeral building block of an instance.
```
Comment (dhiller) [inline docs/architecture.md]:
```suggestion
VirtualMachineInstance (VMI) is the custom resource that represents the basic ephemeral building block of an instance.
```

**Comment by ravidbro** [comment]
@dhiller I think that you need to override again after I fixed the spelling

**Pull request merged**